http/basic/src/main/java/org/wildfly/security/http/basic/BasicAuthenticationMechanism.java

```diff
@@ -169,8 +170,7 @@ public void evaluateRequest(final HttpServerRequest request) throws HttpAuthenti
                             } else {
                                 httpBasic.debugf("User %s authorization failed.", username);
                                 fail();
-
-                                request.authenticationFailed(httpBasic.authorizationFailed(username), response -> prepareResponse(request, displayRealmName, response));
+                                request.authenticationFailed(httpBasic.authorizationFailed(username), response -> response.setStatusCode(HttpConstants.FORBIDDEN));
```

Comment:
@keshav-725 Just a minor, the commit that adds this change should be the same (have a same commit hash) like the affiliated PR's commit - https://github.com/wildfly-security/wildfly-elytron/pull/1899 . You can do that by rebasing this on that PR's branch or by cherry-picking that commit in this branch.